### DISCUSSION TITLE
Again on group action- for the  Firmware/cloud password

### DISCUSSION THREAD

**Discussion by frank777777** [JurajNyiri/HomeAssistant-Tapo-Control#669] [Discuss]
Two months ago, I proposed a small 'group action' to express our frustration over the lack of response from TAPO, despite Juraj’s excellent work. Unfortunately, it seems that only a few have followed this initiative.

If Juraj still hasn’t received a response, it means we need to intensify our efforts and make our voices heard more. I encourage everyone to participate  and kindly  share your opinions on the following platforms: App Stores, Trustpilot, Amazon (for product reviews of items you’ve purchased), and the Tapo app. I guarantee company are sensitive to feedback,  and I'm expecting if the numbers are big,  they are going to  reach Juraj swift-

Only by uniting our voices can we make a difference!
Frank

PS:  if you need some ideas, here below my messages  to Tapo" 

_"I purchased this webcam with great enthusiasm, mainly because it was compatible with Home Assistant through a GitHub - HACS integration. for sure not a original feature, but a good  compromise. The idea of having a device perfectly integrated into my smart home ecosystem was very appealing, and initially, everything seemed to work wonderfully.

Unfortunately, this satisfaction was short-lived. For over two month now, the APIs necessary for the webcam to function with Home Assistant have been unavailable, rendering the device completely unusable within my home automation system.

This situation is made even more frustrating by the lack of clear information or timelines for resolving the issue. The developer of the HACS plugin, on which many users rely, is also waiting for responses that don’t come, leaving both developers and end users in limbo.

The lack of communication from the company and the absence of support for a solution that I chose precisely for its compatibility with Home Assistant are unacceptable for me. even if it wasn't on a feature list. I feel compelled to advise against purchasing this webcam for anyone using or planning to use Home Assistant, unless the API issue is resolved and the promised functionality is restored."_

**Comment by disprosio** [comment]
Thanks Frank,
I followed your advice and sent that message as feedback on their app. Today I got a reply back from them:

>  Thank you for contacting TP-Link Spain technical support.
> 
> The C520WS camera is not advertised as compatible with Home Assistant, therefore we cannot guarantee that Home Assistant will work, and we cannot give support for it.
> 
> Best regards
> 
> TP-Link España - Dpt Soporte Técnico

I think it was pretty much clear in the text above that the integration was not original and none is expecting them to support it.

**Comment by blanknam3d** [comment]
To be honest, I firmly believe they're gonna do the right thing... just that they'll take an unusually long time to do it

For a good long while, people were requesting the ability to manually set network settings on the cameras, so they'd always use a static IP themselves instead of relying on DHCP IP+MAC binding. 

I especially had wanted such a feature, because every now and then for some reason one or the other would fight to use the same address, even though they had static IPs set in my router's config... and like 3 or so years after other people initially complained, all of a sudden they added the feature!

So I'm inclined to believe that they *do* listen to the users, and they *do* plan to do the right thing... just that they've got it on the furthest-back back-burner, and although it's definitely on the to-do list, they're handling things that more directly affect the average user's experience first.

That being said I hope they pick up the pace on this one. I ultimately reverted the firmware of my Tapo C200 cameras to use the integration, and now have to put up with the issue I had that was solved in 1.3.13 - in the meantime, I have to work around that issue by rebooting my AP and extenders all at the same time.

**Reply by blanknam3d** [reply]
Just saw the most recent update: Glad to see TP-Link make good with the users, even if it took a while - looks like I can confidently go order a C520WS now to replace that one poorly-behaving Yi outdoor camera I have!

**Comment by mistraloz** [comment]
The easy thing that we can do are : 
1/ Changing the material : TP Link is not reliable, we can beg for their autorisation but we are the customers, we buy it... and we can not use it because of the lock in behaviours. 
2/ Inform other peoples who want to buy there material about the lock in (i did it on amazon and play store). 
3/ Inform other peoples about the security issue : TP Link said that is not possible to publish the interoperability code because that is a security issue... So if it's a security issue to analyze the protocole, that should be true (and in addition, that is possible because @JurajNyiri said that the token does not refresh : that is a very bad implementation of the OWASP recommandations) 

On my side i will not spend too many time with that, i will return and use another. 

PS: I am totally aligned with JuraNyiri strategy, that is the best approach to ensure the viability of the integration... but for users, that is just not valid to wait for that one day the supplier will grant us a favor and allow us to use what we baught. What they answered to @disprosio is not true : the lock in is not a feature that a provider can implement as he wants (at least in EU/UE). Class action for this amount of money is just overkill so nevermind...

**Comment by DanielLaberge** [comment]
They might not care about existing customers, but I'm sure they care about potential new customers!

I suggest everyone start tweeting at them. I just did: https://x.com/labergedev/status/1852552680054407299?s=46&t=vC-P_J5IkDEptyjn6xGMxA

I found out about the broken Home Assistant support JUST before pulling the trigger on buying a bunch of these cameras based on the good reviews and (admittedly outdated) recommendations.

TP-Link if you're reading this: you just lost the sale of a dozen Tapo C225s in favour of Anker's Eufy. Get your house in order.

**Reply by ddonevski** [reply]
I'm afraid that the big picture is that threads on this repo have around 60 participants. I use Tapo cams in my weekend house in a small village in Croatia. My neighbor also has Tapo cams and uses the official app. He doesn't even consider the possibility of putting the effort into installing Home Assistant, Frigate and this integration. He's an average user. My point is, there are many Tapo cams in random little villages all over the world. A few hundred users who would like to integrate 40 EUR cams with third party apps so we don't have to buy several hundred EUR cams with similar functionality developed by camera manufacturers aren't really important. If they allow this some time in the future, it will be very generous. But it's not their priority. If more people were like users in this thread, able to set up their cams with third party apps, allowing it would be against manufacturer's interest. Since this is a small community, it wouldn't affect them, but it isn't their priority and we can consider it a gift. I'm not sure that a small group of users, who wish to use thier product in a way manufacturer didn't intend it to be used, protesting against them has any effect and it may piss them off.

**Comment by partytimeexcellent** [comment]
I'm a new Tapo camera owner. Very disappointed in the RTSP implementation. Constant errors that brings down my Frigate install. Then I learn about the Tapo cloud integration that seems promising...nope, no longer works. I'm just gonna send this camera back and get another Dahua. So tired of trying to save a little money and ending up with endless troubleshooting. I will send TP Link a more detailed version of this message detailing why I'm returning the camera and why I'll leave them negative reviews everywhere.

**Reply by blanknam3d** [reply]
Awfully late to see this, but: I run two Tapo C200s myself. You sure you didn't have some issue with Wi-FI link speed?

I used to have very frequent RTSP-related issues with mine, until finding out that my AP handling at least 70 devices was somehow connected on only a 100 megabit link... - they *really* do not like streaming over a saturated/poor network link, as I've found by how two TVs & a computer was all it took to make them unreliable. Simply fixing that has since gotten them running perfectly for me.
